File: src/business/rule/engine/serviceprovider/BookProductBusinessRuleProvider.java
package business.rule.engine.serviceprovider;

import business.rule.engine.model.Order;
import business.rule.engine.service.IUserService;

public class BookProductBusinessRuleProvider implements IBusinessRuleProvider{
	private IUserService userService;

    public BookProductBusinessRuleProvider(IUserService userService)
    {
        this.userService = userService;
    }

    public String NameRequirement = "BookProduct";

    // If the payment is for a book, create a duplicate packing slip for the royalty department.
    public boolean Process(Order order)
    {
        if (order.Product.getName() == NameRequirement)
        {
            if (order.PackingSlip!=null && order.PackingSlip.contains("royalty"))
            {
                return true;
            }
            order.PackingSlip += "royalty";
            return true;
        }

        return false;
    }
}

File: src/BusinessRuleProviders/BookProductBusinessRuleProvider.cs
using Raymaker.RulesEngine.Application.Model;

namespace Raymaker.RulesEngine.Application.BusinessRuleProviders
{
    public class BookProductBusinessRuleProvider : IBusinessRuleProvider
    {
        private readonly IUserService userService;

        public BookProductBusinessRuleProvider(IUserService userService)
        {
            this.userService = userService;
        }

        public string NameRequirement => "BookProduct";

        // If the payment is for a book, create a duplicate packing slip for the royalty department.
        public (bool isSatisfied, string message) Process(Order order)
        {
            if (order.Product.GetType() == typeof(Book))
            {
                if (!string.IsNullOrEmpty(order.PackingSlip) && 
                    order.PackingSlip.Contains("royalty"))
                {
                    return (true, $"{NameRequirement}: Royalty slip already added");
                }
                order.PackingSlip += "royalty";
                return (true, $"{NameRequirement}: Added royalty slip");
            }

            return (false, $"{NameRequirement}: No action");
        }
    }
}
